api/types/clusterconfig.go

```diff
@@ -322,16 +334,73 @@ func (c *ClusterConfigV3) CheckAndSetDefaults() error {
 		c.Spec.KeepAliveCountMax = int64(defaults.KeepAliveCountMax)
 	}
 
+	// Detect invalid or inconsistent overrides.
+	if err := c.keepOnlyValidOverrides(); err != nil {
+		return trace.Wrap(err)
+	}
+
 	return nil
 }
 
+// keepOnlyValidOverrides checks for inconsistencies with the current override
+// profile, and resets inadmissible overrides so that they do not impact the
+// value of the corresponding ClusterConfig field.
+func (c *ClusterConfigV3) keepOnlyValidOverrides() error {
+	var errs []error
+	if !c.HasOverride() {
+		return nil
+	}
+
+	if c.Override.SessionRecording != "" {
+		var err error
+		allowedOverrides := []string{RecordAtNode, RecordAtNodeSync, RecordOff}
+		if !utils.SliceContainsStr(allowedOverrides, c.Override.SessionRecording) {
```

Comment:
This should probably be a part of the `ClusterOverride`'s own `Check()` method.

Reply:
I decided to write it this way for two reasons:
1. Since `ClusterConfigOverride` doesn't get stored in the backend, it should really be seen as just a thin key-value holder.
2. I wanted to emphasize the fact that the `allowedOverrides` is constraint for both the `ClusterConfigOverride` value and the `ClusterConfig` value that need to be present for the override to be applicable.